src/prefect/_internal/compatibility/deprecated.py

```diff
@@ -32,6 +34,7 @@
 M = TypeVar("M", bound=BaseModel)
 T = TypeVar("T")
 
+_AcceptableDate: TypeAlias = Optional[Union[datetime.datetime, str]]
```

Comment:
nit: I think a comment saying that a `datetime` is strongly preferred would be helpful.